graph/articulationpoints.go

```diff
@@ -0,0 +1,62 @@
+// Algorithm to identify articulation points in a connected graph
+// You can read more about articulation points here: https://en.wikipedia.org/wiki/Biconnected_component
+// and the implementation and explanation here: https://cptalks.quora.com/Cut-Vertex-Articulation-point
+
+package graph
+
+import "github.com/TheAlgorithms/Go/math/min"
+
+var (
+	time               int    // time variable to keep track of the time of discovery_time of a vertex
+	is_ap              []bool // boolean array to mark articulation points
+	visited            []bool // boolean array to mark visited vertices
+	child_cnt          []int  // array to store the number of children of a vertex
+	discovery_time     []int  // array to store the time of discovery of a vertex
+	earliest_discovery []int  // array to store the earliest discovered vertex reachable from a vertex
+)
+
+func ArticulationPoint(graph Graph) []bool {
+	//initialize all the variables
+	time = 0
+	is_ap = make([]bool, graph.vertices)
+	visited = make([]bool, graph.vertices)
+	child_cnt = make([]int, graph.vertices)
+	discovery_time = make([]int, graph.vertices)
+	earliest_discovery = make([]int, graph.vertices)
+
+	ArticulationPointHelper(graph, 0, -1)
+
+	if child_cnt[0] == 1 {
+		is_ap[0] = false // if the root has only one child, it is not an articulation point
+	}
+
+	return is_ap
+}
+
+func ArticulationPointHelper(graph Graph, u int, parent int) {
```

Comment:
Make it unexported - unlikely to be used other than your own function. Exporting it pollutes the API.

```suggestion
func articulationPointHelper(graph Graph, u int, parent int) {
```